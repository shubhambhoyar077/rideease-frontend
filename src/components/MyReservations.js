import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchReservations,
  cancelReservation,
} from '../redux/reservations/reservationsSlice';
import '../styles/MyReservations.css';

function MyReservations() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchReservations());
  }, [dispatch]);

  const reservations = useSelector((state) => state.reservations.reservations);
  const handleCancelReservation = (reservationId) => {
    dispatch(cancelReservation(reservationId));
  };

  return (
    <section className="res-container">
      <div>
        <h1>My Reservations</h1>
      </div>
      <div className="res-details">
        {reservations.map((reservation) => (
          <div key={reservation.reservation.id}>
            <div className="reservation">
              <div className="res-texts">
                <h5>{reservation.service.name}</h5>
                <p>
                  City:
                  {reservation.reservation.city}
                </p>
                <p>
                  Date:
                  {reservation.reservation.date}
                </p>
                <button
                  type="button"
                  className="cancel-btn"
                  onClick={() => handleCancelReservation(reservation.reservation.id)}
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        ))}
      </div>
    </section>
  );
}

export default MyReservations;
