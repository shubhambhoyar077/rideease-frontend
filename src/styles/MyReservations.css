.container {
  margin-left: 0%;
}

@media (max-width: 1100px) {
  .container {
    margin-top: 0%;
    margin-left: 0%;
  }

  .card {
    display: flex;
    flex-direction: row;
  }
}
