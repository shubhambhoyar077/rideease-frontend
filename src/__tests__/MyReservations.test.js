import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import MyReservations from '../components/MyReservations';

const mockStore = configureMockStore([]);

describe('MyReservations Component', () => {
  it('should render correctly', () => {
    const store = mockStore({
      reservations: {
        reservations: [
          {
            service: { name: 'Test Service', image: 'test.jpg' },
            reservation: { id: 1, city: 'Test City', date: '2023-07-15' },
          },
        ],
      },
    });
    const tree = renderer.create(
      <Provider store={store}>
        <MyReservations />
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
