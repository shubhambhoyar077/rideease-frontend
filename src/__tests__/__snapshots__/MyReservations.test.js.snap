// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReservations Component should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="text-center"
  >
    <h1
      style={
        Object {
          "padding": "2%",
        }
      }
    >
      My Reservations
    </h1>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-11 mb-4"
    >
      <div
        className="card flex-md-row"
      >
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Test Service
          </h5>
          <p
            className="card-text"
          >
            City:
             
            Test City
          </p>
          <p
            className="card-text"
          >
            Date:
             
            2023-07-15
          </p>
          <button
            className="btn btn-danger"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </div>
        <img
          alt="Test Service"
          className="card-img-right rounded-circle"
          src="test.jpg"
          style={
            Object {
              "width": "20%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
